app/lib/common/providers/room_providers.dart

```diff
@@ -210,3 +211,38 @@ final roomIsMutedProvider =
   return (await ref.watch(roomNotificationStatusProvider(roomId).future)) ==
       'muted';
 });
+
+class MemberNotFound extends Error {}
+
+class RoomNotFound extends Error {}
+
+class RoomMemberQuery {
+  final String roomId;
+  final String userId;
+  const RoomMemberQuery(this.roomId, this.userId);
+}
+
+final roomMemberProvider = FutureProvider.autoDispose
+    .family<ProfileData, RoomMemberQuery>((ref, query) async {
+  final room = await ref.watch(maybeRoomProvider(query.roomId).future);
+  if (room == null) {
+    throw RoomNotFound;
+  }
+  final member = await room.getMember(query.userId);
+  return ref.watch(userProfileDataProvider(member).future);
```

Comment:
yes, because displayname and avatar are both `async` and thus we need to `await` for them to be finished. the provider takes care of wrapping all that (and as a bonus for the same member objects also caches it nicely and is thus faster on subsequent requests compared to doing `getProfile` locally).